### PR TITLE
update versions to 8.0.4

### PR DESCRIPTION
**Pull request by danmoseley**
Update to yesterday's 8.0.4 where we shipped one.

The other updates came along for the ride. Used this procedure https://github.com/dotnet/aspire/pull/3512#issue-2232592285
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3590)

### CODE REVIEW

**Review by drewnoakes — APPROVED** [review]

**Comment by danmoseley** [comment]
oops .. need to mirror

**Comment by danmoseley** [comment]
https://github.com/dotnet/aspire/pull/3591#issuecomment-2048974949

**Comment by danmoseley** [comment]
Same Linux failures

File: eng/Versions.props [hunk 1]
```diff
@@ -36,15 +36,15 @@
...
     <MicrosoftAspNetCoreOpenApiPackageVersion>8.0.4</MicrosoftAspNetCoreOpenApiPackageVersion>
-    <MicrosoftAspNetCoreOutputCachingStackExchangeRedisPackageVersion>8.0.3</MicrosoftAspNetCoreOutputCachingStackExchangeRedisPackageVersion>
-    <MicrosoftExtensionsCachingStackExchangeRedisPackageVersion>8.0.3</MicrosoftExtensionsCachingStackExchangeRedisPackageVersion>
-    <MicrosoftExtensionsDiagnosticsHealthChecksEntityFrameworkCorePackageVersion>8.0.3</MicrosoftExtensionsDiagnosticsHealthChecksEntityFrameworkCorePackageVersion>
-    <MicrosoftExtensionsDiagnosticsHealthChecksPackageVersion>8.0.3</MicrosoftExtensionsDiagnosticsHealthChecksPackageVersion>
-    <MicrosoftExtensionsFeaturesPackageVersion>8.0.3</MicrosoftExtensionsFeaturesPackageVersion>
-    <MicrosoftEntityFrameworkCoreCosmosPackageVersion>8.0.3</MicrosoftEntityFrameworkCoreCosmosPackageVersion>
+    <MicrosoftAspNetCoreOutputCachingStackExchangeRedisPackageVersion>8.0.4</MicrosoftAspNetCoreOutputCachingStackExchangeRedisPackageVersion>
+    <MicrosoftExtensionsCachingStackExchangeRedisPackageVersion>8.0.4</MicrosoftExtensionsCachingStackExchangeRedisPackageVersion>
```
Comment (danmoseley) [inline eng/Versions.props:43]:
```suggestion
    <MicrosoftExtensionsCachingStackExchangeRedisPackageVersion>8.0.3</MicrosoftExtensionsCachingStackExchangeRedisPackageVersion>
```

**Comment by danmoseley** [comment]
this is hitting the same SIGABRT.

**Comment by danmoseley** [comment]
started #3610 to help bisect

**Comment by eerhardt** [comment]
@joperezr caught the Redis failures last night. There was a change in 8.0.4 that broke our tests: https://github.com/dotnet/aspnetcore/pull/54239. 

The test is:

https://github.com/dotnet/aspire/blob/58ceb834c15595e20083f327a1791e9997e3be1c/tests/Aspire.StackExchange.Redis.DistributedCaching.Tests/DistributedCacheConformanceTests.cs#L61-L72

Checking for OTel tracing from doing a "GetAsync" on the cache. With the change to aspnetcore we are seeing 2 extra "FireAndForget" "CLIENT" traces come through that the test isn't expecting.

**Comment by danmoseley** [comment]
is there a bug for the redis caching issue meantime?

File: eng/Versions.props [hunk 2]
```diff
@@ -14,19 +14,21 @@
...
     <MicrosoftDotnetSdkInternalPackageVersion>8.0.100-rtm.23512.16</MicrosoftDotnetSdkInternalPackageVersion>
+    <!-- DCP -->
     <MicrosoftDeveloperControlPlanedarwinamd64PackageVersion>0.2.2</MicrosoftDeveloperControlPlanedarwinamd64PackageVersion>
     <MicrosoftDeveloperControlPlanedarwinarm64PackageVersion>0.2.2</MicrosoftDeveloperControlPlanedarwinarm64PackageVersion>
     <MicrosoftDeveloperControlPlanelinuxamd64PackageVersion>0.2.2</MicrosoftDeveloperControlPlanelinuxamd64PackageVersion>
     <MicrosoftDeveloperControlPlanelinuxarm64PackageVersion>0.2.2</MicrosoftDeveloperControlPlanelinuxarm64PackageVersion>
     <MicrosoftDeveloperControlPlanewindows386PackageVersion>0.2.2</MicrosoftDeveloperControlPlanewindows386PackageVersion>
     <MicrosoftDeveloperControlPlanewindowsamd64PackageVersion>0.2.2</MicrosoftDeveloperControlPlanewindowsamd64PackageVersion>
     <MicrosoftDeveloperControlPlanewindowsarm64PackageVersion>0.2.2</MicrosoftDeveloperControlPlanewindowsarm64PackageVersion>
+    <!-- Other -->
```
Comment (danmoseley) [inline eng/Versions.props:25]: marker to avoid conflicts with DCP updates

File: tests/Aspire.EndToEnd.Tests/IntegrationServicesFixture.cs [hunk 3]
```diff
@@ -80,7 +80,7 @@ public Task DumpComponentLogsAsync(TestResourceNames resource, ITestOutputHelper
     {
         if (resource == TestResourceNames.None)
         {
-            return;
+            return Task.CompletedTask;
```
Comment (danmoseley) [inline tests/Aspire.EndToEnd.Tests/IntegrationServicesFixture.cs:83]: build fix

Comment (eerhardt) [reply tests/Aspire.EndToEnd.Tests/IntegrationServicesFixture.cs:83]: How'd this get in?

Comment (radical) [reply tests/Aspire.EndToEnd.Tests/IntegrationServicesFixture.cs:83]: oooh! conflicting changes in my PRs!! crap! apologies!

**Comment by eerhardt** [comment]
> is there a bug for the redis caching issue meantime?

I’ll open one tomorrow

**Pull request merged**